feat: scrollable report page (BAL-3550)

r4zendev opened the pull request:
<img width="1155" alt="image" src="https://github.com/user-attachments/assets/7b1328c5-8643-43b0-b79a-1e201f688a7d" />
<img width="1165" alt="image" src="https://github.com/user-attachments/assets/586bf05e-7b97-412c-a6a6-fab582679b61" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced an interactive report interface with a navigable sidebar, smooth scrolling, and dynamic section highlighting.
	- Added a custom hook for managing and rendering report sections.
	- Enhanced handling of demo accounts with conditional UI elements and tooltips.
- **Refactor**
	- Simplified report display by replacing tab-based navigation with a direct view of report content.
	- Enhanced flexibility for risk indicator and summary components.
	- Encapsulated column definitions within a custom hook for dynamic behavior based on account type.
- **Style**
	- Streamlined layouts and spacing across components for improved responsiveness.
- **Chores**
	- Reorganized code structure and standardized type declarations to enhance maintainability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

r4zendev committed:
wip

r4zendev committed:
wip

r4zendev committed:
adaptive credibility

r4zendev committed:
wip use-report-sections

r4zendev committed:
scrollable report sections

MatanYadaev commented on packages/ui/src/components/templates/report/components/BusinessReportSummary/BusinessReportSummary.tsx:
```diff
@@ -1,33 +1,37 @@
   summary: string;
   ongoingMonitoringSummary?: string;
-  sections: ReadonlyArray<{
+  riskIndicators: ReadonlyArray<{
```
Why rename?

rzssh replied:
from the name `sections` in this component it's very hard to understand the purpose of this prop, so i renamed it to better represent the intent. do you think there could be any issues with it?

MatanYadaev commented on packages/ui/src/components/templates/report/components/BusinessReportSummary/BusinessReportSummary.tsx:
```diff
@@ -1,33 +1,37 @@
-    search: string;
-    riskIndicators: z.infer<typeof RiskIndicatorSchema>[] | null;
+    search?: string;
+    indicators: Array<z.infer<typeof RiskIndicatorSchema>> | null;
```
Why rename?

r4zendev committed:
fix: remove redundant code

r4zendev committed:
fix: imports

r4zendev committed:
fix

alonp99 commented on apps/backoffice-v2/src/domains/business-reports/components/BusinessReport/BusinessReport.tsx:
```diff
@@ -0,0 +1,151 @@
+  const [isSidebarOpen, setIsSidebarOpen] = useState(true);
+
+  return (
+    <div ref={parentRef} className={`flex transition-all duration-300`}>
```
How does parentRef used?

shanegrouber committed:
Bal 3611 - UI changes (#3094)

* refactor(CaseOverview): improve tab generation logic

- Replace inline tab logic with a dedicated function
- Ensure 'ubos' case returns the correct tab name
- Maintain existing functionality for other domain tab names

* refactor(CaseOverview): simplify tab generation logic

- Remove unnecessary getTab function
- Use camelCase on the domain string directly

* feat(MerchantMonitoringBusinessReport): update button label for navigation

- Change button text from "Back" to "View All Reports"
- Improve clarity on navigation purpose for users

* feat(scripts): enhance auto-commit script with specific target file option

- Add command line argument to specify target file for git diff
- Update git commit command to include the target file in commit

* feat(MerchantMonitoring): add conditional alert column to the table

- Introduce 'showAlertColumn' prop to toggle alert column visibility
- Modify MerchantMonitoringTable and related components accordingly
- Update logic to filter columns based on 'showAlertColumn' setting

* feat(MerchantMonitoring): update merchant monitoring logic to include demo flag

- Replace showAlertColumn with isDemo in merchant monitoring logic
- Modify MerchantMonitoringTable to conditionally render alert column based on isDemo
- Adapt useColumns to use isDemo for column visibility control

* refactor(MerchantMonitoring): rename isDemo to isDemoAccount

- Update variable and prop names for clarity
- Ensure consistent naming across components and hooks

* style(sidebar): improve class ordering for consistency

- Reorder class names in Sidebar components for better readability
- Make minor formatting adjustments across several Sidebar and TextEditor components
- Maintain consistent styling practices throughout the codebase

* refactor(report-tabs): streamline report tab configurations

- Remove unnecessary social media and ecosystem analysis tabs
- Simplify the report tabs structure for better clarity
- Ensure the remaining tabs maintain functional integrity

* feat(ui): enhance merchant monitoring for demo accounts

- Add tooltip messages indicating feature limitations for demo accounts
- Disable monitoring action for trial users with appropriate feedback
- Update logic to check account type before rendering specific options

* fix(MerchantMonitoring): update button icons and styles in MerchantMonitoring page

- Replace Table2 icon with Layers for better clarity
- Update button text for Upload Multiple Merchants to Batch Actions
- Refactor button styles to improve visibility and user experience

* feat(MerchantMonitoringBusinessReport): align Options dropdown with website domain

- Implement dropdown menu for monitoring options
- Add confirmation dialog for deboarding merchants
- Integrate feedback mechanism via form fields for user reasons

* refactor(merchant-monitoring): restructure merchant monitoring components

- Convert columns definition to a memoized hook for performance enhancement
- Standardize access to demo account configuration across components
- Improve button label for better clarity on navigation functionality

---------

Co-authored-by: Alon Peretz <8467965+alonp99@users.noreply.github.com>
